### PR TITLE
NXP-24815: filter empty entry in directory json writer

### PR DESCRIPTION
**Pull request by michaelva**

### CODE REVIEW

**Comment by guirenard** [comment]
You should run a T&P as some unit tests may fail with your change. I launched https://qa.nuxeo.org/jenkins/job/TestAndPush/job/ondemand-testandpush-grenard-maven3/216/

**Comment by guirenard** [comment]
:+1:

File: nuxeo-services/nuxeo-platform-directory/nuxeo-platform-directory-core/src/main/java/org/nuxeo/ecm/directory/io/DirectoryEntryJsonWriter.java [hunk 1]
```diff
@@ -127,7 +128,7 @@ protected void writeEntityBody(DirectoryEntry entry, JsonGenerator jg) throws IO
                 Property property = document.getProperty(fieldName.getPrefixedName());
                 boolean managed = false;
                 Object value = property.getValue();
-                if (value != null && value instanceof String) {
+                if (value instanceof String && StringUtils.isNotEmpty((String)value)) {
```
Comment (driverpt) [inline nuxeo-services/nuxeo-platform-directory/nuxeo-platform-directory-core/src/main/java/org/nuxeo/ecm/directory/io/DirectoryEntryJsonWriter.java]: Possible `NullPointerException here`

**Comment by driverpt** [comment]
👍

**Pull request merged**